blog/cross-account-eventbridge.md

```diff
@@ -0,0 +1,129 @@
+
+The term is [commonly used](https://kgb1001001.github.io/cloudadoptionpatterns/Event-Based-Architecture/Event-Backbone/) for such systems based on Apache Kafka, since Kafka was one of the first technologies that enabled event backbones for microservice communication with massive scale and performance. Since Kafka was first released over a decade ago, cloud managed services have evolved to the degree where you don't need to Kafka to have a scalable, reliable event backbone. Amazon EventBridge is the most obvious example, since it has managed to pull off the amazing feat of having a large feature set and massive scalability while remaining one of the simplest cloud services there is.
+
+If you are a Kafka fan, there is effort being put in by AWS in reducing the complexity with the managed MSK service and a generally available 'serverless' version in the works too. I would compare MSK to EventBridge in the same way I would compare EKS to Fargate or Lambda. You get a lot more control and configurability but even with the AWS managed service, you still have plenty of complexity.
+
+The beauty of something like EventBridge is that the investment is so low. If your needs evolve, you can adapt and use alternative options for specific cases. You are not stuck with it because of a large investment in infrastructure or training. Need durability? Add SQS! Need low-latency, ordered streams? Add Kinesis! It's possible to build a event backbone on Kinesis or SNS/SQS but EventBridge is still the best place to start, integrates with more services and has really good cross-account support.
+
+## Cross-account EventBridge
+
+We already mentioned that EventBridge has good support for cross-account scenarios. With EventBridge, you can create a Rule with any other EventBridge _bus_ as a target. This bus can be in a different account.
+
+![Cross account EventBridge](./cross-account-eb.png)
+
+For this to work, the target bus must have a policy that allows the source account to send events to it.
+
+Now, let's imagine this idea at a larger scale, where we have multiple accounts, each with their own applications or services. Where does each application need to send events? There are a few options here. If you want to explore all the options, take a look at [this great talk from re:Invent 2020 on Building event-driven architectures](https://youtu.be/Wk0FoXTUEjo).  In this article, I'll focus on my preferred option, referred to in that video as the "single-bus*, multi account-pattern". There are are in fact multiple buses, but a central bus in a dedicated account is used to route messages to multiple accounts, each with their own local bus.
+
+- Every service send events to a global bus, a dedicated bus in a separate account
+- Every service receives events from a local bus in its own account
+- The global bus has rules to route all events to every local bus except the local bus of the event sender
```

Comment:
Would it make sense that this is an implementation of the _fan-out_ pattern?

Reply:
It could implement the fan out pattern if a single message is routed to multiple local busses but you could argue that since we're not delivering to a pool of identical workers this isn't quite the same as fan-out. More of a routing topology.

Reply:
Hmmm. Good points. It might be good to describe this as a fan-out because it might help the reader grasp the concept. From the global bus's perspective, every local bus is the same and it is _fanning out_ to multiple local buses.